signbank/dictionary/csv_import.py

```diff
@@ -231,10 +231,23 @@ def import_nzsl_share_gloss_csv(request):
         for row in glossreader:
             if glossreader.line_num == 1:
                 continue
-            if row["id"] not in existing_nzsl_share_ids:
-                new_glosses.append(row)
+            if row["id"] in existing_nzsl_share_ids:
+
+                # nzsl_share_id is not a reliable index, due to manual intervention
+                try:
+                    gloss = Gloss.objects.filter(nzsl_share_id=row["id"]).get()
+                except (Gloss.DoesNotExist, Gloss.MultipleObjectsReturned) as e:
+                    print(e)
+                    continue
+
+                # if gloss has video/s we skip it, otherwise we add it anyway
+                if gloss.glossvideo_set.all().exists():
+                    skipped_existing_glosses.append(row)
```

Comment:
question: am I right in thinking this means the entry will get _completely_ reimported?

i.e. if you have a entry with a particular name whose video goes missing, and then you reimport it with a different name, will that result in both the video being "fixed" _and_ the name being updated?

I don't think that's a bad thing, but it's worth confirming I think before we land this and maybe even mentioning to the client, just to head off possible unexpected surprises.

Reply:
Client has confirmed that we don't want to completely reimport the signs in this case, we just want to reimport their videos:

> We do NOT want entries to be completely re-imported - the entries that were previously imported without video may have already been modified in Signbank. We only want the video to be added to those records with an NZSL_Share_ID that do not yet have a video.

Reply:
Ah, Ok, that's a chunk more work then. In fact it's almost a completely new import mechanism.  I'll have to think about it.

Reply:
Yup, there are a number of ways, but it does mean teasing that import loop apart is the thing.
Obviously I'm going to have to do so.

Reply:
Ah, looks like joke was on me/us anyway. I failed to spot yesterday that that was creating duplicate entries. I'd have had to shore up the import process anyway. 😆

Reply:
Closing this thread prior to pushing.
New code only imports the video and prevents the rest of the import for the gloss from running.